code/game/supplyshuttle.dm

```diff
@@ -125,99 +125,112 @@ var/datum/controller/supply/supply_controller = new()
-		return 1
-	user.set_interaction(src)
-	var/dat = "<body>"
+/obj/structure/machinery/computer/supply_drop_console/ui_status(mob/user)
+	. = ..()
+	if(inoperable(MAINT))
+		return UI_CLOSE
 
-	if(can_pick_squad)
-		if(!current_squad) //No squad has been set yet. Pick one.
-			dat += "Current Squad: <A href='?src=\ref[src];operation=pick_squad'>----------</A><BR>"
-		else
-			dat += "Current Squad: [current_squad.name] Squad</A><A href='?src=\ref[src];operation=pick_squad'>\[Change\]</A><BR>"
+/obj/structure/machinery/computer/supply_drop_console/ui_state(mob/user)
+	return GLOB.not_incapacitated_and_adjacent_state
 
-	dat += "<BR><B>Supply Drop Control</B><BR><BR>"
-	if(!current_squad)
-		dat += "No squad selected!"
-	else
-		dat += "<B>Current Supply Drop Status:</B> "
-		var/cooldown_left = (current_squad.supply_cooldown + drop_cooldown) - world.time
-		if(cooldown_left > 0)
-			dat += "Launch tubes resetting ([round(cooldown_left/10)] seconds)<br>"
-		else
-			dat += SET_CLASS("Ready!", INTERFACE_GREEN)
-			dat += "<br>"
-		dat += "<B>Launch Pad Status:</b> "
-		var/obj/structure/closet/crate/C = locate() in current_squad.drop_pad.loc
-		if(C)
-			dat += SET_CLASS("Supply crate loaded", INTERFACE_GREEN)
-			dat += "<BR>"
-		else
-			dat += "Empty<BR>"
-		dat += "<B>Longitude:</B> [x_supply] <A href='?src=\ref[src];operation=supply_x'>\[Change\]</a><BR>"
-		dat += "<B>Latitude:</B> [y_supply] <A href='?src=\ref[src];operation=supply_y'>\[Change\]</a><BR><BR>"
-		dat += "<A href='?src=\ref[src];operation=dropsupply'>\[LAUNCH!\]</a>"
-	dat += "<BR><BR>----------------------<br>"
-	dat += "<A href='?src=\ref[src];operation=refresh'>{Refresh}</a><br>"
-
-	show_browser(user, dat, "Supply Drop Console", "overwatch", "size=550x550")
-	return
+/obj/structure/machinery/computer/supply_drop_console/tgui_interact(mob/user, datum/tgui/ui)
+	. = ..()
+	ui = SStgui.try_update_ui(user, src, ui)
+	if (!ui)
+		ui = new(user, src, "SupplyDropConsole", name)
+		ui.open()
 
-/obj/structure/machinery/computer/supply_drop_console/Topic(href, href_list)
-	if(..())  //Checks for power outages
-		return
-	if(!href_list["operation"])
+/obj/structure/machinery/computer/supply_drop_console/ui_data(mob/user)
+	. = ..()
+	check_pad()
+	.["busy"] = busy
+	.["can_pick_squad"] = can_pick_squad
+	.["current_squad"] = current_squad
+	.["launch_cooldown"] = drop_cooldown
+	.["next_fire"] = COOLDOWN_TIMELEFT(src, next_fire)
+	.["x_offset"] = x_supply
+	.["y_offset"] = y_supply
+	.["active"] = busy
+	.["loaded"] = loaded_crate
+	.["crate_name"] = loaded_crate.name
+
+/obj/structure/machinery/computer/supply_drop_console/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+	. = ..()
+	if(.)
 		return
-	switch(href_list["operation"])
+
+	switch(action)
+		if("set_x")
+			var/new_x = text2num(params["set_x"])
+			if(!new_x)
+				return
+			x_supply = new_x
+
+		if("set_y")
+			var/new_y = text2num(params["set_y"])
+			if(!new_y)
+				return
+			y_supply = new_y
```

Comment:
do some sanity validation here for positive and in bounds, though just 0 <= x < 255 is probably fine

Reply:
not done - the values input here are obfuscated so they can be negative and out of bounds with relation to real coords

actual sanitisation takes place in the launch proc

	var/x_coord = deobfuscate_x(x_supply)
	var/y_coord = deobfuscate_y(y_supply)
	var/z_coord = SSmapping.levels_by_trait(ZTRAIT_GROUND)
	if(length(z_coord))
		z_coord = z_coord[1]
	else
		z_coord = 1 // fuck it

	var/turf/T = locate(x_coord, y_coord, z_coord)
	if(!T)
		to_chat(usr, "[icon2html(src, usr)] [SPAN_WARNING("Error, invalid coordinates.")]")
		return

as such I want to keep it the same as the old system and not sanitise values here, so that if invalid coords are given then req can't tell.